### DISCUSSION TITLE
.env variables are undefined in production

### DISCUSSION THREAD

**Discussion by rianfloo** [Shopify/shopify-app-template-node#757] [Q&A] [answered]
Hello :) 

I developed a plugin and try to submit to review but I've got a Javascript error in production mode. 

It works perfectly in development but not in production mode. 

Here is the log I've got on Heroku:

```
2022-02-14T17:46:30.225357+00:00 app[web.1]: SCOPES: process.env.SCOPES.split(","),
2022-02-14T17:46:30.225357+00:00 app[web.1]: ^
2022-02-14T17:46:30.225358+00:00 app[web.1]:
2022-02-14T17:46:30.225358+00:00 app[web.1]: TypeError: Cannot read properties of undefined (reading 'split')
```

My .env is filled. 
When I console.log process.env, I've got and object but its not containing SCOPES and other variables inside my .env file. 

Any ideas why?

Thanks a looot! :)

**Comment by Michael-Gibbons** [comment, marked as answer]
@rianfloo This doesn't sound like an issue with this package, when you switch to production in heroku you need to specify the config variables using either the heroku cli or the heroku admin.

https://devcenter.heroku.com/articles/config-vars
